test/topology_custom/test_mv_read_concurrency.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import logging
+
+from test.topology.conftest import skip_mode
+from test.pylib.util import wait_for_view
+from cassandra import ReadTimeout, WriteTimeout
+
+logger = logging.getLogger(__name__)
+
+# This test verifies that the writes causing view updates don't impact the latency of regular reads
+# due to contention on the read concurrency semaphore.
+# The test creates a table and a table with a materialized view, and then runs a large number of writes causing view updates
+# while concurrently running a small read workload on the other table.
+# The test fails if any of the reads times out.
+# Reproduces https://github.com/scylladb/scylladb/issues/8873
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_mv_read_concurrency(manager: ManagerClient) -> None:
+    node_count = 1
+    # Disable cache to make reads use the read concurrency semaphore.
+    # Tests remove the rcs multiplier by default, here we use a slightly smaller one (1 instead of default 2) to hit the issue faster.
+    servers = await manager.servers_add(node_count, config={'enable_tablets': True, 'enable_cache': False, 'reader_concurrency_semaphore_serialize_limit_multiplier': 1})
+
+    cql, _ = await manager.get_ready_cql(servers)
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}"
+                        "AND tablets = {'initial': 2}")
```

Comment:
Why do you want to explicitly enable tables, and moreover - choose 2 tables explicitly - instead of just using the default?

Reply:
I started testing with just 1 tablet and changed it to 2 because otherwise the read table and write (base) table would just be on different shards and there would be no interference between view update reads and regular reads.
But there's actually no reason to use 2 tablets instead of the default so I removed this in the rebase.
I assume we want to focus on testing with tablets rather than without in new features. This shouldn't really matter in this test but should I test with vnodes instead here or make the test run with both options?